tool/src/kotlin/mod.rs

```diff
@@ -1932,7 +1991,113 @@ mod test {
             let type_name = opaque_def.name.to_string();
             // test that we can render and that it doesn't panic
             let (_, result) =
-                ty_gen_cx.gen_opaque_def(opaque_def, &type_name, "dev.gigapixel", "somelib");
+                ty_gen_cx.gen_opaque_def(opaque_def, &type_name, "dev.gigapixel", "somelib", false);
+            insta::assert_snapshot!(result)
+        }
+    }
+
+    #[test]
+    fn test_opaque_gen_with_finalizers() {
```

Comment:
thought: this test doesn't need to be that complex if it's only testing one thing.

Reply:
good point